lang/py/src/avro/io.py

```diff
@@ -119,11 +120,27 @@ def validate(expected_schema, datum):
       return isinstance(datum, Decimal)
     return isinstance(datum, str)
   elif schema_type == 'int':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
+      if expected_schema.logical_type == 'date':
+        return isinstance(datum, datetime.date)
+      elif expected_schema.logical_type == 'time-millis':
```

Comment:
`time-millis` type is referenced from avro spec https://avro.apache.org/docs/1.8.0/spec.html#Time+%28millisecond+precision%29
i think it is better to maintain type consistency